src/tribler/core/components/torrent_checker/torrent_checker/torrent_checker.py

```diff
@@ -281,159 +290,156 @@ def get_valid_trackers_of_torrent(self, torrent_id):
+                'infohash': infohash_hex,
+                'num_seeders': 0,
+                'num_leechers': 0,
+                'last_tracker_check': int(time.time()),
+                'health': 'updated'
+            })
+
+    def _create_session_for_request(self, tracker_url, timeout=20) -> Optional[TrackerSession]:
+        self._logger.debug(f'Creating a session for the request: {tracker_url}')
+
+        required_hops = self.config.download_defaults.number_hops
+        actual_hops = len(self.socks_listen_ports or [])
+        if required_hops > actual_hops:
+            self._logger.warning(f"Dropping the request. Required amount of hops doesn't reached. "
+                                 f'Required hops: {required_hops}. Actual hops: {actual_hops}')
             return None
-        proxy = ('127.0.0.1', self.socks_listen_ports[hops - 1]) if hops > 0 else None
+        proxy = ('127.0.0.1', self.socks_listen_ports[required_hops - 1]) if required_hops > 0 else None
         session = create_tracker_session(tracker_url, timeout, proxy, self.socket_mgr)
-
-        if tracker_url not in self._session_list:
-            self._session_list[tracker_url] = []
-        self._session_list[tracker_url].append(session)
-
-        self._logger.debug("Session created for tracker %s", tracker_url)
+        self._logger.info(f'Tracker session has been created: {session}')
+        self._sessions[tracker_url].append(session)
         return session
 
     async def clean_session(self, session):
-        self.tracker_manager.update_tracker_info(session.tracker_url, not session.is_failed)
+        url = session.tracker_url
+
+        self.tracker_manager.update_tracker_info(url, not session.is_failed)
         # Remove the session from our session list dictionary
-        self._session_list[session.tracker_url].remove(session)
-        if len(self._session_list[session.tracker_url]) == 0 and session.tracker_url != "DHT":
-            del self._session_list[session.tracker_url]
+        self._sessions[url].remove(session)
+        if len(self._sessions[url]) == 0 and url != DHT:
+            del self._sessions[url]
 
         await session.cleanup()
+        self._logger.debug('Session has been cleaned up')
 
-    async def _on_result_from_session(self, session, result_list):
-        await self.clean_session(session)
-        # FIXME: this should be probably handled by cancel, etc
-        if self._should_stop:
-            return
-
-        return result_list
-
-    def _update_torrent_result(self, response):
-        infohash = response['infohash']
-        seeders = response['seeders']
-        leechers = response['leechers']
-        last_check = response['last_check']
-
-        self._logger.debug("Update result %s/%s for %s", seeders, leechers, hexlify(infohash))
-
+    def update_torrent_health(self, health: InfohashHealth):
+        self._logger.info(f'Update torrent health: {health}')
```

Comment:
This method should not update health if the information is incorrect, for example (and most importantly) when the "last_check" time is future. Because of time drift, it is possible to have unsynchronized time between machines, so we should accept records that are, for example, just a few seconds in the future, but the tolerable difference should not be too large.

Reply:
Sometimes the torrent was checked locally just recently, but other peers have incorrect health records with big numbers of seeders for this torrent and continue spreading these records. When such health records arrive on our machine, we should ignore them to avoid overriding the correct health. So, even if incoming health information looks correct, the local node should ignore it if recently it already checks this torrent and has fresh health info about it.